warehouse/macaroons/caveats/__init__.py

```diff
@@ -50,8 +51,7 @@
 
 
 @as_caveat(tag=0)
-@dataclass(frozen=True)
-class Expiration(Caveat):
+class Expiration(Caveat, frozen=True):
```

Comment:
Curious - is there anything preventing us from shipping this refactor independently of other changes? Looks like a nice cleanup.